src/result_gen_with_api.py

```diff
@@ -411,37 +443,247 @@ def prepare_scan_weight_result_object(
+            api : objects
+                object of api_endpoints class
+            scan_metadata_path : str
+                path to store scan metadata
+        """
+        self.api = api
+        self.scan_metadata_path = scan_metadata_path
+
+    def get_unprocessed_scans(self):
+        """
+        Gets unprocessed_scans from api and returns the no of scans
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        None
+        """
+
+        return self.api.get_scan(self.scan_metadata_path)
+
+    def get_scan_metadata(self):
+        with open(self.scan_metadata_path, 'r') as f:
+            scan_metadata_obj = json.load(f)
+        scan_metadata = scan_metadata_obj['scans'][0]
+
+        return scan_metadata
+
+
+class PrepareArtifacts:
+    """
+    A class to prepare artifacts for result generation.
+
+    ...
+
+    Attributes
+    ----------
+    api : object
+        object of api_endpoints class
+    scan_metadata : json
+        metadata of the scan to run weight flow on
+    scan_parent_dir : str
+        directory where scans are stored
+
+    Methods
+    -------
+    download_artifacts(input_format):
+        Download artifacts for the scan
+    check_artifact_format():
+        Checks the format of the artifact
+    add_artifacts_to_format_dictionary():
+        Sort artifacts according to input format
+    process_scan_metadata():
+        Process artifacts in a scan.
+    create_scan_dir():
+        Create directory to store artifacts in scan.
+    create_artifact_dir():
+        Create directory to store downloaded artifacts.
+    """
+    def __init__(self, api, scan_metadata, scan_parent_dir):
+        self.api = api
+        self.scan_metadata = scan_metadata
+        self.format_wise_artifact = {}
+        self.scan_parent_dir = scan_parent_dir
+        self.scan_dir = os.path.join(
+            self.scan_parent_dir,
+            self.scan_metadata['id'])
+
+    def download_artifacts(self, input_format):
+        print(f"\nDownloading Artifacts for { input_format } format")
+        self.artifacts = []
+
+        for i, artifact in enumerate(
+                self.format_wise_artifact[input_format]):
+            mod_artifact = copy.deepcopy(artifact)
+
+            print("\nDownloading Artifact Name: ", mod_artifact["file"], '\n')
+            status_code = self.api.get_files(
+                mod_artifact["file"], os.path.join(self.scan_dir, input_format))
+            # status_code = get_files_mockup(mod_artifact["file"], format_dir)
+            if status_code == 200:
+                mod_artifact['download_status'] = True
+                self.artifacts.append(mod_artifact)
+
+        print(f"\nBelow Artifacts for { input_format } workflow\n")
+        print(self.artifacts)
+        print("\nDownload Artifact for completed\n")
+
+        return self.artifacts
+
+    def check_artifact_format(self, format):
+        if format == 'image/jpeg' or format == 'rgb':
```

Comment:
I like shorter code ;)
```suggestion
        if format in ['image/jpeg', 'rgb']:
```